src/net_processing.cpp

```diff
@@ -349,7 +349,16 @@ struct CNodeState {
 
     TxDownloadState m_tx_download;
 
-    CNodeState(CAddress addrIn, std::string addrNameIn) : address(addrIn), name(addrNameIn) {
+    //! Whether this peer is an inbound connection
+    bool m_is_inbound;
+
+    //! Whether this peer is a manual connection
+    bool m_is_manual_connection;
+
+    CNodeState(CAddress addrIn, std::string addrNameIn, bool is_inbound, bool is_manual) :
+        address(addrIn), name(std::move(addrNameIn)), m_is_inbound(is_inbound),
```

Comment:
Any particular reason why this isn't just `CNodeState(.., const std::string& addrNameIn, ...) : ..., name(addrNameIn)`?

Reply:
> Any particular reason why this isn't just CNodeState(.., const std::string& addrNameIn, ...) : ..., name(addrNameIn)?

This would be worse because you can't move from a const reference and there would be an unnecessary copy.

If a function is inserting an argument into a data structure, it's good practice for it to take the argument by value or by rvalue reference instead of by const reference, to avoid pointless copying.

Reply:
I am pretty sure there will be no copying if the input is discardable, except for `-O0`, but I could be mistaken.

Reply:
Initializing the member either has to copy or move, and you can't move from a const object, so the copy constructor will be called. If you are referring to `-O0` it sounds like you are thinking about [copy elision](https://en.cppreference.com/w/cpp/language/copy_elision), but that doesn't apply here for member initialization.

Simple advice is to use `const&` for an argument that you want to copy from, `&&` for an argument that you want to move from, and pass by value when you want to give the caller a choice of whether to copy or move.

Reply:
I see. But we are doing `const type& var` all over the place elsewhere. This is in fact why I wondered about the "derivation from the standard" here. (A simple search for `const std::string&` gives 799 results.)

Reply:
Passing a string by reference doesn't need a copy. Using it to initialize a member does.

Reply:
Interesting. Sorry for clogging up the PR.